mindmeld/components/entity_resolver.py

```diff
@@ -569,31 +738,242 @@ def _predict_exact_match(self, entity):
+            )
+            transformer_model = Transformer(self._sbert_model_pretrained_name_or_abspath)
+            pooling_model = Pooling(transformer_model.get_word_embedding_dimension(),
+                                    pooling_mode_cls_token=_encoding_type == "cls",
+                                    pooling_mode_max_tokens=False,
+                                    pooling_mode_mean_tokens=_encoding_type == "mean",
+                                    pooling_mode_mean_sqrt_len_tokens=False)
+            modules = [transformer_model, pooling_model]
+            self._sbert_model = SentenceTransformer(modules=modules)
+        except OSError:
+            logger.error("Could not initialize the model name through huggingface models; "
+                         "Resorting to model names in sbert.net. "
+                         "Input 'encoding_type', if provided is thus ignored")
+            self._sbert_model = SentenceTransformer(self._sbert_model_pretrained_name_or_abspath)
+
+        # load mappings.json data
+        entity_map = self.resource_loader.get_entity_map(self.type)
+        self._exact_match_mapping = self._process_entity_map(
+            self.type, entity_map
+        )
 
-        except EsConnectionError as e:
-            logger.error(
-                "Unable to connect to Elasticsearch: %s details: %s", e.error, e.info
+        # load embeddings for this data
+        cache_path = self.cache_path(self.pretrained_name)
+        if clean and os.path.exists(cache_path):
+            os.remove(cache_path)
+        if not self._lazy_resolution and os.path.exists(cache_path):
+            self._load()
+            self.dirty = False
+        else:
+            synonyms = [*self._exact_match_mapping["synonyms"]]
+            synonyms_encodings = self._encode(synonyms)
+            self._preloaded_mappings_embs = dict(zip(synonyms, synonyms_encodings))
+            self.dirty = True
+
+        if self.dirty and not self._lazy_resolution:
+            self._dump()
+
+    def _predict(self, entity):
+        """Predicts the resolved value(s) for the given entity using cosine similarity.
+
+        Args:
+            entity (Entity, tuple): An entity found in an input query, or a list of n-best entity \
+                objects.
+
+        Returns:
+            (list): The top 20 resolved values for the provided entity.
+        """
+
+        syn_embs = self._preloaded_mappings_embs
+        entity = entity[0]  # top_entity
+        entity_emb = self._encode(entity.text)[0]
+
+        try:
+            sorted_items = self._sort_using_cosine_dist(syn_embs, entity_emb)
+            values = []
+            for synonym, score in sorted_items:
+                cnames = self._exact_match_mapping["synonyms"][synonym]
+                for cname in cnames:
+                    for item in self._exact_match_mapping["items"][cname]:
+                        item_value = copy.copy(item)
```

Comment:
Is it appropriate to do a `copy.deepcopy` here?

Reply:
I think it might not be required